sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -325,39 +325,27 @@ def _buildprovider(
     else:
         user_agent = USER_AGENT
 
-    clients = []
-    if connection_string and endpoint:
-        clients.append(
-            ConfigurationClientWrapper.from_connection_string(
-                endpoint, connection_string, user_agent, retry_total, retry_backoff_max, **kwargs
-            )
-        )
-        failover_endpoints = find_auto_failover_endpoints(endpoint, replica_discovery_enabled)
-        for failover_endpoint in failover_endpoints:
-            failover_connection_string = connection_string.replace(endpoint, failover_endpoint)
-            clients.append(
-                ConfigurationClientWrapper.from_connection_string(
-                    failover_endpoint, failover_connection_string, user_agent, retry_total, retry_backoff_max, **kwargs
-                )
-            )
-        provider._replica_client_manager.set_clients(clients)
-        return provider
-    if endpoint and credential:
-        clients.append(
-            ConfigurationClientWrapper.from_credential(
-                endpoint, credential, user_agent, retry_total, retry_backoff_max, **kwargs
-            )
-        )
-        failover_endpoints = find_auto_failover_endpoints(endpoint, replica_discovery_enabled)
-        for failover_endpoint in failover_endpoints:
-            clients.append(
-                ConfigurationClientWrapper.from_credential(
-                    endpoint, credential, user_agent, retry_total, retry_backoff_max, **kwargs
-                )
-            )
-        provider._replica_client_manager.set_clients(clients)
-        return provider
-    raise ValueError("Please pass either endpoint and credential, or a connection string with a value.")
+    interval: int = kwargs.get("refresh_interval", 30)
+    if interval < 1:
+        raise ValueError("Refresh interval must be greater than or equal to 1 second.")
+
+    min_backoff: int = kwargs.get("min_backoff", 30) if kwargs.get("min_backoff", 30) <= interval else interval
```

Comment:
Does it make sense to rather use `Math.min(kwargs.pop("min_backoff", 30), interval)` here for readability?